java/src/jmri/InstanceManager.java

```diff
@@ -63,12 +63,14 @@
  * @author Bob Jacobsen Copyright (C) 2001, 2008, 2013, 2016
  * @author Matthew Harris copyright (c) 2009
  */
-public class InstanceManager {
+public final class InstanceManager {
 
-    protected static final HashMap<Class<?>, ArrayList<Object>> managerLists = new HashMap<>();
-    private static final InstanceInitializer initializer = new jmri.managers.DefaultInstanceInitializer();
+    // the default instance of the InstanceManager
+    private static volatile InstanceManager defaultInstanceManager = null;
```

Comment:
Would it be better to have 
```
private static volatile InstanceManager defaultInstanceManager = new InstanceManager();
```
so that `getDefault()` doesn't have to check each time?